packages/nc-gui-v2/components/dashboard/settings/userManagement/ShareBase.vue

```diff
@@ -0,0 +1,221 @@
+<script setup lang="ts">
+import { useToast } from 'vue-toastification'
+import OpenInNewIcon from '~icons/mdi/open-in-new'
+import { dashboardUrl } from '~/utils/urlUtils'
+
+import MdiReload from '~icons/mdi/reload'
+import DownIcon from '~icons/ic/round-keyboard-arrow-down'
+import ContentCopyIcon from '~icons/mdi/content-copy'
+import MdiXmlIcon from '~icons/mdi/xml'
+import { copyTextToClipboard } from '~/utils/miscUtils'
+const toast = useToast()
+
+interface ShareBase {
+  uuid?: string
+  url?: string
+  role?: string
+}
+
+enum ShareBaseRole {
+  Editor = 'editor',
+  Viewer = 'viewer',
+}
+
+const { $api, $e } = useNuxtApp()
+let base = $ref<null | ShareBase>(null)
+const showEditBaseDropdown = $ref(false)
+const { project } = useProject()
+
+const url = $computed(() => (base && base.uuid ? `${dashboardUrl()}#/nc/base/${base.uuid}` : null))
+
+const loadBase = async () => {
+  try {
+    if (!project.value.id) return
+
+    const res = await $api.project.sharedBaseGet(project.value.id)
+    base = {
+      uuid: res.uuid,
+      url: res.url,
+      role: res.roles,
+    }
+  } catch (e) {
+    console.error(e)
+    toast.error('Something went wrong')
+  }
+}
+
+const createShareBase = async (role = ShareBaseRole.Viewer) => {
+  try {
+    if (!project.value.id) return
+
+    const res = await $api.project.sharedBaseUpdate(project.value.id, {
+      roles: role,
+    })
+
+    base = res || {}
+    base.role = role
+  } catch (e) {
+    console.error(e)
+    toast.error('Something went wrong')
```

Comment:
Similarly, cast e to any and print the error using await extractSdkResponseErrorMsg(e)